metals/src/main/scala/scala/meta/internal/metals/StacktraceAnalyzer.scala

```diff
@@ -62,6 +62,13 @@ class StacktraceAnalyzer(
     } yield makeGotoLocationCodeLens(location, range)).toSeq
   }
 
+  def workspaceFileLocationFromLine(line: String): Option[l.Location] = {
+    fileLocationFromLine(line).filter { location =>
+      val path = location.getUri().toAbsolutePath
+      path.startWith(workspace) && !path.isInReadonlyDirectory(workspace)
```

Comment:
> path.startWith(workspace)

This won't be true for single files.

Reply:
Ach right, changed to check if we are in a jar filesystem